components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/JWTTokenIssuer.java

```diff
@@ -622,10 +622,11 @@ protected JWTClaimsSet createJWTClaimSet(OAuthAuthzReqMessageContext authAuthzRe
         JWTClaimsSet jwtClaimsSet;
 
         // Handle custom claims
+        // TODO : We need to handle jwt access token claims issuing for old and new application.
         if (authAuthzReqMessageContext != null) {
             jwtClaimsSet = handleCustomClaims(jwtClaimsSetBuilder, authAuthzReqMessageContext);
         } else {
-            jwtClaimsSet = handleCustomClaims(jwtClaimsSetBuilder, tokenReqMessageContext);
+            jwtClaimsSet = handleJWTAccessTokenOIDCClaims(jwtClaimsSetBuilder, tokenReqMessageContext);
```

Comment:
Are we planning to invoke the prior implementation within this or switching the implementation from here.
IMO, it's better to have a one claim handler for AT and handle the legacy behaviour within. So in any other implementations it can be referred

Reply:
Yes, we plan to do that. This is only for the testing.